profiles/models.py

```diff
@@ -84,27 +148,42 @@ class Affiliation(db.Model):
     _city = db.Column(db.Text(), name='city', nullable=False)
     _region = db.Column(db.Text(), name='region')
     _country = db.Column(ISO3166Country, name='country', nullable=False)
-    starts = db.Column(UTCDateTime, nullable=False)
-    ends = db.Column(UTCDateTime)
+    starts = composite(Date, '_starts_year', '_starts_month', '_starts_day')
+    _starts_year = db.Column(db.Integer(), name='starts_year', nullable=False)
+    _starts_month = db.Column(db.Integer(), name='starts_month')
+    _starts_day = db.Column(db.Integer(), name='starts_day')
+    _ends = composite(Date, '_ends_year', '_ends_month', '_ends_day')
+    _ends_year = db.Column(db.Integer(), name='ends_year')
+    _ends_month = db.Column(db.Integer(), name='ends_month')
+    _ends_day = db.Column(db.Integer(), name='ends_day')
     restricted = db.Column(db.Boolean(), nullable=False)
     profile_id = db.Column(db.String(ID_LENGTH), db.ForeignKey('profile.id'))
     profile = db.relationship('Profile', back_populates='affiliations')
     position = db.Column(db.Integer())
 
     # pylint: disable=too-many-arguments
-    def __init__(self, affiliation_id: str, address: Address, organisation: str, starts: datetime,
-                 department: str = None, ends: datetime = None, restricted: bool = False) -> None:
+    def __init__(self, affiliation_id: str, address: Address, organisation: str, starts: Date,
+                 department: str = None, ends: Date = None, restricted: bool = False) -> None:
         self.id = affiliation_id
         self.department = department
         self.organisation = organisation
         self.address = address
-        self.starts = pendulum.timezone('utc').convert(starts)
-        self.ends = pendulum.timezone('utc').convert(ends) if ends else None
+        self.starts = starts
+        self._ends = ends
         self.restricted = restricted
 
+    @property
+    def ends(self) -> Optional[Date]:
+        # SQLAlchemy appears not to allow nullable composites columns.
+        if self._ends and self._ends.year:
+            return self._ends
+
+    def set_ends(self, ends: Optional[Date]) -> None:
+        self._ends = ends
+
     def is_current(self) -> bool:
-        starts = pendulum.instance(self.starts)
-        ends = pendulum.instance(self.ends) if self.ends else None
+        starts = pendulum.instance(self.starts.lowest_possible())
+        ends = pendulum.instance(self.ends.highest_possible()) if self.ends else None
```

Comment:
Had wondered about adding `is_past()` and `is_future()` (and `is_today()`) to `Date`, but given the ambiguity around incomplete dates might be best to just leave here.

Reply:
I agree, the incomplete dates will make that more work than its probably worth at the moment.